src/sparseml/recipe_template/utils.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Optional
+
+from sparseml.pytorch.sparsification import (
+    ACDCPruningModifier,
+    GMPruningModifier,
+    MagnitudePruningModifier,
+    Modifier,
+    QuantizationModifier,
+)
+from sparseml.sparsification import ModifierYAMLBuilder
+
+
+__all__ = [
+    "ModifierBuildInfo",
+]
+
+
+class ModifierBuildInfo:
+    """
+    A class with state and helper methods for building a recipe from
+    modifier(s)
+
+    :param modifier: class of the Modifier in question
+    :param modifier_name: str A string name to be prefixed with all attribute
+        variables defined for this modifier, defaults to "mod"
+    :param fields: An Optional dictionary mapping modifier attribute names to their
+        corresponding values
+    """
+
+    def __init__(
+        self,
+        modifier: Modifier,
+        modifier_name: str = "mod",
+        fields: Optional[Dict[str, Any]] = None,
+    ):
+        self.modifier = modifier
+        self.modifier_name = modifier_name
+        self.modifier_builder = ModifierYAMLBuilder(modifier)
+        self.fields = fields or {}
+        self.__modifier_recipe_variables = {}
+        self.update(updated_fields=self.fields)
+
+    def update(self, updated_fields: Optional[Dict[str, Any]] = None):
+        """
+        Utility method to update the fields for current modifier
+
+        :param updated_fields: An Optional dictionary mapping modifier attribute names
+            to their corresponding values
+        """
+        for key, value in updated_fields.items():
+            variable_name = f"{self.modifier_name}_{key}"
+            setattr(self.modifier_builder, key, f"eval({variable_name})")
+            self.__modifier_recipe_variables[variable_name] = updated_fields[key]
+
+    @property
+    def modifier_recipe_variables(self) -> Dict[str, Any]:
+        """
+        :returns: A dict mapping b/w recipe variable names --> values
+            for the current object
+        """
+        return self.__modifier_recipe_variables
+
+
+# PRUNING MODIFIERS INFO
+
+_PRUNING_MODIFIER_INFO_REGISTRY = {
```

Comment:
All those registers do not have any relationship to `ModifierBuildInfo`, nor are they designed to be exported from this script. Maybe we should move them to a separate script?

Reply:
Ok, not a separate script, but I would probably bring them to the top of the script and make them public (why are they private? there are not used in the code by `ModifierBuildInfo`) and add them to `__all__`

Reply:
+1 they seem to be used in #1054 , so i think they should be moved there (and honestly could probably just be functions instead of global dicts)

Reply:
I would prefer to keep them in the current dictionary form or as formattable strings. It's easier to read and the functionality/intention is more straightforward